### ISSUE TITLE
BUG/BENCH: clough_tocher benchmark fails

### ISSUE THREAD

**Issue by h-vetinari** [scipy/scipy#18860]
It seems we're not picking up deprecation warnings or even ValueError/TypeError from the benchmarks, causing them to bitrot.

Even a recent passing run contains the [following](https://circleci.com/api/v1.1/project/github/scipy/scipy/71191/output/104/0?file=true&allocation-id=64ad73cd3db5641253f87c09-0-build%2F68590B51) upon closer inspection:

```
[  9.63%] ··· ...clough_tocher failed
[  9.63%] ··· ================ =========== ========= ======= ========== ========
               max_coord_size   n_samples   flipped    tol    max_iter          
              ---------------- ----------- --------- ------- ---------- --------
                     10             10         1      0.001      1       failed 
                     10             10         1      0.001      10      failed 
                     10             10         1      0.001     100      failed 
                     10             10         1      0.001     400      failed 
                     10             10         1      1e-06      1       failed 
                     10             10         1      1e-06      10      failed 
                     10             10         1      1e-06     100      failed 
[and so on]
```
with stack trace:
```
               For parameters: 10, 10, 1, 0.001, 100
               /home/circleci/repo/benchmarks/benchmarks/interpolate.py:374: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
                 interpolate.CloughTocher2DInterpolator.__init__(self, points, None, tol=tol, maxiter=maxiter)
               TypeError: only size-1 arrays can be converted to Python scalars
               
               The above exception was the direct cause of the following exception:
               
               Traceback (most recent call last):
                 File "/home/circleci/.pyenv/versions/3.9.17/lib/python3.9/site-packages/asv/benchmark.py", line 1184, in main_run_server
                   main_run(run_args)
                 File "/home/circleci/.pyenv/versions/3.9.17/lib/python3.9/site-packages/asv/benchmark.py", line 1052, in main_run
                   skip = benchmark.do_setup()
                 File "/home/circleci/.pyenv/versions/3.9.17/lib/python3.9/site-packages/asv/benchmark.py", line 583, in do_setup
                   result = Benchmark.do_setup(self)
                 File "/home/circleci/.pyenv/versions/3.9.17/lib/python3.9/site-packages/asv/benchmark.py", line 515, in do_setup
                   setup(*self._current_params)
                 File "/home/circleci/repo/benchmarks/benchmarks/interpolate.py", line 422, in setup
                   self.interp = CloughTocherInterpolatorValues(
                 File "/home/circleci/repo/benchmarks/benchmarks/interpolate.py", line 374, in __init__
                   interpolate.CloughTocher2DInterpolator.__init__(self, points, None, tol=tol, maxiter=maxiter)
                 File "interpnd.pyx", line 945, in scipy.interpolate.interpnd.CloughTocher2DInterpolator.__init__
                 File "interpnd.pyx", line 79, in scipy.interpolate.interpnd.NDInterpolatorBase.__init__
               ValueError: setting an array element with a sequence.
```

There are other failures as well, but this one is the most egregious

**Comment by h-vetinari** [comment]
@hovavalon, I see you committed these benchmarks in https://github.com/scipy/scipy/commit/3c54130362bfdcf93622144ad4b8b63277482831, but I don't think they ever ran correctly. Trying to reproduce the class setup locally, I get something like
```
>>> pprint(points)
[array([0.10446416, 0.1221474 , 0.3137688 , 0.36109471, 0.45491803,
       0.50983691, 0.59643396, 0.65044056, 0.81275552, 0.91155261]),
 array([0.10209738, 0.52098389, 0.66941367, 0.70426217, 0.87207028])]
>>> points[0].shape
(10,)
>>> points[1].shape
(5,)
```

I believe this gets passed verbatim down to `NDInterpolatorBase.__init__`, which then
https://github.com/scipy/scipy/blob/cfe80116aaa145061246b8aec0e98248fbefb835/scipy/interpolate/interpnd.pyx#L76
calls
https://github.com/scipy/scipy/blob/cfe80116aaa145061246b8aec0e98248fbefb835/scipy/interpolate/interpnd.pyx#L158

and the failure happens somewhere in there. I cannot tell if `self.points` should be a tuple, or where the logic goes wrong, but I believe this was uncaught because our benchmark suite was not failing on errors.

**Comment by hovavalon** [comment]
Hello!
These benchmarks were indeed added by me, and they indeed never ran correctly.
At the time of the PR's approval there were global errors in the ASV benchmark so the benchmark itself wasn't tested.
In the PR's discussion there are working code examples for the feature benchmarked here:
https://github.com/scipy/scipy/pull/18376

I will be able to submit a PR fixing the benchmark now that the benchmark suite is working, but it might take me some time to get to it, in the meantime you can disable the benchmark or try using the code examples from the PR to fix them yourself.

Sorry for the inconvenience

**Comment by h-vetinari** [comment]
@ev-br, it seems you had a closer look at #18376 at the time - do you think you could help fix the broken benchmarks? It's not clear to me from that PR how the committed-as-broken benchmarks are supposed to be adapted from the examples in that PR.

Background: I want to unblock https://github.com/scipy/scipy/pull/19052, and currently that consists of deleting the failing benchmarks, which I'd like to avoid if possible.

**Comment by ev-br** [comment]
I'll take a look.

**Comment by ev-br** [comment]
@h-vetinari : see https://github.com/scipy/scipy/pull/19104

**Comment by h-vetinari** [comment]
Thanks a lot @ev-br!

@hovavalon, if you want to propose a less invasive change to the benchmarks, please let us know ASAP. Of course, you can always come back later and iterate on top of #19104, but I'm looking to get this unblocked in a matter of days, not weeks.

**Comment by hovavalon** [comment]
I think the current change is OK for now, maybe in the future I will have the benchmark updated.

**Comment by h-vetinari** [comment]
OK, thanks @hovavalon. If you want to track this in an issue, feel free to open a new one. I'm closing this one as solved by #19104.
